packages/paste-core/primitives/combobox/stories/mutliselect.stories.tsx

```diff
@@ -32,102 +36,105 @@ export const BasicMultiCombobox: StoryFn = () => {
-        addSelectedItem(selected);
-      }
-      // ℹ️ This action from the dropdown library is strongly typed to accept only a string
-      //   to reset, we need to pass "null" so we will cast it as a string e.g. "null as string".
+    inputValue,
+    items: getFilteredItems(items, selectedItems, inputValue),
+    onStateChange: ({inputValue: currentInputValue = '', type, selectedItem}) => {
+      switch (type) {
+        case useComboboxPrimitive.stateChangeTypes.InputChange:
+          setInputValue(currentInputValue);
+
+          break;
+        case useComboboxPrimitive.stateChangeTypes.InputKeyDownEnter:
+        case useComboboxPrimitive.stateChangeTypes.ItemClick:
+        case useComboboxPrimitive.stateChangeTypes.InputBlur:
+          if (selectedItem) {
+            setInputValue('');
+            addSelectedItem(selectedItem);
+
+            // ℹ️ This action from the dropdown library is strongly typed to accept only a string
+            //   to reset, we need to pass "null" so we will cast it as a string e.g. "null as string".
 
-      // ℹ️ Downshift does not clear the "selectedItem" state after a selection has changed
-      //   with the multiple selection implementation, we need to force this case, so that if
-      //   we remove the item, it isn't still marked as "Selected" in the combobox state.
-      selectItem((null as unknown) as string);
+            // ℹ️ Downshift does not clear the "selectedItem" state after a selection has changed
+            //   with the multiple selection implementation, we need to force this case, so that if
+            //   we remove the item, it isn't still marked as "Selected" in the combobox state.
+            selectItem((null as unknown) as string);
+          }
+
+          break;
+        default:
+          break;
+      }
     },
   });
 
   const inputId = seed('multi-select-primitive-input');
 
-  const dropdownProps = getDropdownProps({
-    preventKeyAction: isOpen,
-    ...getToggleButtonProps({tabIndex: 0, id: inputId}),
-  });
-
   return (
     <>
       <Box marginBottom="space40" position="relative">
-        <Label {...getLabelProps({htmlFor: inputId})}>Choose a component</Label>
-        <Box {...getComboboxProps({role: 'combobox'})}>
-          <Input
-            type="text"
-            {...getInputProps({
-              ...dropdownProps,
-            })}
-            value={inputValue}
-          />
+        <Label htmlFor={inputId} {...getLabelProps()}>
+          Choose a component
+        </Label>
+
+        <Box {...getComboboxProps({role: 'combobox'})} display="flex">
+          <Box display="flex" flexGrow={1}>
+            <Input
+              {...getInputProps(getDropdownProps({preventKeyAction: isOpen, type: 'text', id: inputId}))}
+              flexGrow={1}
+            />
+          </Box>
+          <Box display="flex" flexGrow={0}>
+            <Button {...getToggleButtonProps({tabIndex: 0})} aria-label="toggle-button">
```

Comment:
For a11y: aria-label should be a readable string that describes what the button does, like "Expand combobox" or something

Reply:
Could we move this to a separate ticket and address for all the combobox examples? I can do a jira-less PR too, just want to prevent scope creep for this one.

Reply:
Yee - if its in other examples lets make another ticket.